e2e/pages/adf/content-services/documentListPage.ts

```diff
@@ -0,0 +1,66 @@
+/*!
+ * @license
+ * Copyright 2019 Alfresco Software, Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { by, element, ElementFinder, browser } from 'protractor';
+import { DataTableComponentPage } from '../dataTableComponentPage';
+import { Util } from '../../../util/util';
+
+export class DocumentListPage {
+
+    rootElement: ElementFinder;
+    actionMenu = element(by.css('div[role="menu"]'));
+    optionButton = by.css('button[data-automation-id*="action_menu_"]');
+    dataTable;
+
+    constructor(rootElement: ElementFinder = element.all(by.css('adf-document-list[data-automation-id="document-list"]')).first()) {
+        this.rootElement = rootElement;
+        this.dataTable = new DataTableComponentPage(this.rootElement);
+    }
+
+    clickOnActionMenu(content) {
+        let row = this.dataTable.getRowParentElement('Display name', content);
+        row.element(this.optionButton).click();
+        Util.waitUntilElementIsVisible(this.actionMenu);
+        browser.sleep(500);
+        return this;
+    }
+
+    dataTablePage() {
+        return new DataTableComponentPage(this.rootElement);
+    }
+
+    getAllRowsNameColumn() {
+        return this.dataTable.getAllRowsColumnValues('Display name');
```

Comment:
Shouldn't have the name of the column here. Could you give it as a parameter?

Reply:
Why ? Since the method is to get the names of every row

Reply:
I was thinking that someone will want to have the exact value in that cell (basically the name of a file/task/etc), but call it differently. For example on Process Services -> Tasks , the column is called 'Name' instead of 'Display name'. Will this method work in that situation?

Reply:
documentListPage.getAllRowsNameColumn() can only be used to get the name of a node in a documentList. For tasks/processes we will have to use dataTablePage.getAllRowsColumnValues('Name') for example, what do you think ?

Reply:
My mistake, i offered an example for datatable :) But Recent Files, that uses documentList, has different column name, it's called 'Name'. So this method will not work in this case.

Reply:
Oh I didn't know that, I'll change it then 👍